src/lib/services/transfer/__tests__/endpoint.spec.ts

```diff
@@ -20,6 +20,44 @@ describe('endpoint', () => {
     }).toMatchSnapshot();
```

Comment:
```
FAIL src/lib/services/transfer/__tests__/endpoint.spec.ts
  ● Test suite failed to run

    src/lib/services/transfer/__tests__/endpoint.spec.ts:28:9 - error TS2[32](https://github.com/globus/globus-sdk-javascript/pull/144/checks#step:6:33)2: Type '{ display_name: string; host_endpoint_id: string; host_path: string; }' is not assignable to type 'Partial<Pick<{ DATA_TYPE: "endpoint"; DATA?: { DATA_TYPE: "server"; id: number; hostname: string | null; port: number | null; scheme: "ftp" | "gsiftp" | null; subject: string | null; incoming_data_port_start: number | null; ... 5 more ...; is_paused: boolean; }[] | undefined; ... 64 more ...; mapped_collection_displ...'.
      Property 'DATA_TYPE' is missing in type '{ display_name: string; host_endpoint_id: string; host_path: string; }' but required in type '{ DATA_TYPE: "shared_endpoint"; }'.

    28         payload: {
               ~~~~~~~

      src/lib/services/transfer/types.ts:220:7
        220       DATA_TYPE: 'shared_endpoint';
                  ~~~~~~~~~
        'DATA_TYPE' is declared here.
    src/lib/services/transfer/__tests__/endpoint.spec.ts:49:11 - error TS2[35](https://github.com/globus/globus-sdk-javascript/pull/144/checks#step:6:36)3: Object literal may only specify known properties, and 'description' does not exist in type 'Partial<Pick<{ DATA_TYPE: "endpoint"; DATA?: { DATA_TYPE: "server"; id: number; hostname: string | null; port: number | null; scheme: "ftp" | "gsiftp" | null; subject: string | null; incoming_data_port_start: number | null; ... 5 more ...; is_paused: boolean; }[] | undefined; ... 64 more ...; mapped_collection_displ...'.

    49           description: 'This is my GCP collection.',
                 ~~~~~~~~~~~
```

If we can safely inject the `DATA_TYPE` for these methods that is an option ([example](https://github.com/globus/globus-sdk-javascript/blob/d1064d938c3c5d288bc504adec907fffb64e9f47/src/lib/services/transfer/service/task-submission.ts#L18)). Basically, if it's static/can only be a single value it's safe to inject for the caller.